ui/src/map/map-store.ts

```diff
@@ -191,16 +200,36 @@ export const mapReducers = {
             }
         });
     },
+    onMissingLinkingVisibilityChange: (
+        state: Map,
+        { payload: visibilitySetting }: PayloadAction<LayerVisibility>,
+    ): void => {
+        state.mapLayers.forEach((layer) => {
+            if (layer.id == visibilitySetting.layerId) {
+                (<LayoutAlignmentsLayer>layer).showMissingLinking = visibilitySetting.visible;
+            }
+        });
+    },
+    onDuplicateTracksVisibilityChange: (
+        state: Map,
+        { payload: visibilitySetting }: PayloadAction<LayerVisibility>,
+    ): void => {
+        state.mapLayers.forEach((layer) => {
+            if (layer.id == visibilitySetting.layerId) {
+                (<LayoutAlignmentsLayer>layer).showDuplicateTracks = visibilitySetting.visible;
+            }
+        });
+    },
```

Comment:
Näiden koodi on käytännössä identtistä tuota muutettavan propsun nimeä lukuunottamatta, ja vastaavia lienee ylempänäkin. Olisikohan näitä mahdollista yhdistää jotenkin jonkinlaiseksi funkkariksi, joka hoitaisi tämän boilerplaten ja jolle annettaisiin vaan visibility-arvo ja vaihdettavan propsun nimi 🤔